x/auth/signing/sign_mode_handler.go

```diff
@@ -30,6 +30,8 @@ type SignerData struct {
 	AccountNumber uint64
 
 	// AccountSequence is the account sequence number of the signer that is used
-	// for replay protection
+	// for replay protection. This field is only useful for Legacy Amino signing,
+	// since in SIGN_MODE_DIRECT the account sequence is already in the signer
+	// info.
 	AccountSequence uint64
```

Comment:
I would have loved to remove `AccountSequence` from SignerData, but afaiu it is still used for SIGN_MODE_LEGACY. So keeping it here.